innovation.game.php

```diff
@@ -22829,7 +22886,28 @@ function stInterInteractionStep() {
                     self::executeDraw($player_id, $value_to_score, 'score');
                     self::executeDraw($player_id, $value_to_score, 'score');
                     break;     
-                
+                // id 413, Echoes age 8: Crossword
+                case "413N1A":
+                    $bonus_counts = self::getAuxiliaryArray();
+                    $remaining_values = count(array_filter($bonus_counts));
+                    // if more than 1 value remains, repeat the choice loop
+                    if($remaining_values > 1) {
+                        $step--;
+                        self::incrementStep(-1);
+                    } else {
+                        // if only one value remains, find the value and draw that many cards
+                        for($i = 0; i < $bonus_counts; $i++) {
```

Comment:
This is a good optimization.  If they are all the same value, then there is no choice to be made.